hack/lint.sh

```
@@ -2,14 +2,14 @@
 set -euo pipefail
 
-# pin golangci-lint version to 1.54.2
-VERSION=v1.54.2
+# pin golangci-lint version
+VERSION=v2.12.0
 if [ "$#" -ne 1 ]; then
     echo "Expected command line argument - container runtime (docker/podman) got $# arguments: $@"
     exit 1
 fi
 
 $1 run --security-opt label=disable --rm -v $(pwd):/app -w /app -e GO111MODULE=on golangci/golangci-lint:${VERSION} \
```

Shellcheck (0.11.0)

[warning] Quote this to prevent word splitting.

(SC2046)